File: src/main/java/pl/training/backend/hotel/model/Reservation.java
package pl.training.backend.hotel.model;

import lombok.Data;
import pl.training.backend.security.entity.User;

import javax.persistence.*;
import java.util.Date;

@Table(name = "reservation")
@Entity
@Data
public class Reservation {


    @Id
    @GeneratedValue
    private Long idReservation;

    @Column(nullable = false)
    private Date startDate;
    @Column(nullable = false)
    private Date endDate;

    public Reservation(Date startDate, Date endDate) {

        this.startDate = startDate;
        this.endDate = endDate;
    }
}

File: client/src/app/hotel/reservation.ts
import {Hotel} from "./hotel";
import {User} from "../security/user";

export class Reservation{

  idReservation: number;
  startDate: Date;
  endDate: Date;
  hotel: Hotel;
  user: User;


  constructor(json) {
    this.idReservation = json['idReservation'];
    this.startDate = json['startDate'];
    this.endDate = json['endDate'];
    this.hotel = json['hotel'];
    this.user = json['user'];
  }
}
